src/psyclone/parse/module_info.py

```diff
@@ -210,3 +309,88 @@ def get_used_symbols_from_modules(self):
             self._extract_import_information()
 
         return self._used_symbols_from_module
+
+    # ------------------------------------------------------------------------
+    def get_psyir(self, routine_name=None):
+        '''Returns the PSyIR representation of this module. This is based
+        on the fparser tree (see get_parse_tree), and the information is
+        cached. If the PSyIR must be modified, it needs to be copied,
+        otherwise the modified tree will be returned from the cache in the
+        future.
+        If the conversion to PSyIR fails, a dummy FileContainer with an
+        empty Container (module) is returned, which avoids additional error
+        handling in many other subroutines.
+        #TODO 2120: This should be revisited when improving on the error
+        handling.
+
+        :param routine_name: optional the name of a routine.
+        :type routine_name: Optional[str]
+
+        :returns: PSyIR representing this module.
+        :rtype: List[:py:class:`psyclone.psyir.nodes.Node`]
+
+        '''
+        if self._psyir is None:
+            try:
+                self._psyir = \
+                    self._processor.generate_psyir(self.get_parse_tree())
+            except (KeyError, SymbolError, InternalError, FortranSyntaxError):
```

Comment:
I have a separate ticket for error handling. I totally agree that we should not ignore errors. But for now this is extremely useful for me, since I often run the kernel extraction across all of LFRic to find errors, and since there are files with 10, 20 or more kernels, it is really useful for me to get the errors printed, but the script keeps on testing ALL kernels at once. I think I have all locations marked with TODO #2120 for this.

Reply:
OK, thanks.